k-distribution/include/kframework/builtin/domains.md

```diff
@@ -465,6 +465,230 @@ module MAP-SYMBOLIC
+and N is the size of the second `RangeMap`). If any ranges are overlapping,
+the value from the second range map overwrites the value in the first for the
+parts where ranges are overlapping. This function is total, which is distinct
+from range map concatenation, a partial function only defined on range maps
+with non overlapping ranges.
+
+```k
+  syntax RangeMap ::= updateRangeMap(RangeMap, RangeMap)            [function, total, hook(RANGEMAP.updateAll)]
+```
+
+### Multiple range map removal
+
+You can remove a `Set` of ranges from a `RangeMap` in O(N*log(M)) time (where
+M is the size of the `RangeMap` and N is the size of the `Set`). For every
+range in the set, all or any part of it that is present in the range map will
+be removed.
+
+```k
+  syntax RangeMap ::= removeAll(RangeMap, Set)            [function, hook(RANGEMAP.removeAll)]
+```
+
+### Range map keys (as `Set`)
+
+You can get a `Set` of all the ranges in a `RangeMap` in O(N) time (where N
+is the size of the `RangeMap`).
+
+```k
+  syntax Set ::= keys(RangeMap)                      [function, total, hook(RANGEMAP.keys)]
+```
+
+### Range map keys (as `List`)
+
+You can get a `List` of all the ranges in a `RangeMap` in O(N) time (where N
+is the size of the `RangeMap`).
+
+```k
+  syntax List ::= "keys_list" "(" RangeMap ")"       [function, hook(RANGEMAP.keys_list)]
+```
+
+### Range map key membership
+
+You can check whether a key is present in a `RangeMap` in O(log(N)) time (where
+N is the size of the `RangeMap`).
+
+```k
+  syntax Bool ::= KItem "in_keys" "(" RangeMap ")"       [function, total, hook(RANGEMAP.in_keys)]
+```
+
+### Range map values (as `List`)
+
+You can get a `List` of all values in a `RangeMap` in O(N) time (where N is the
+size of the `RangeMap`).
+
+```k
+  syntax List ::= values(RangeMap)                   [function, hook(RANGEMAP.values)]
+```
+
+### Range map size
+
+You can get the number of range/value pairs in a `RangeMap` in O(1) time.
+
+```k
+  syntax Int ::= size(RangeMap)                      [function, total, hook(RANGEMAP.size), klabel(sizeRangeMap)]
+```
+
+### Range map inclusion
+
+You can determine whether a `RangeMap` is a strict subset of another `RangeMap`
+in O(M+N) time (where M is the size of the first `RangeMap` and N is the size
+of the second `RangeMap`). Only keys within equal or overlapping ranges that
+are bound to the same value are considered equal.
+
+```k
+  syntax Bool ::= RangeMap "<=RangeMap" RangeMap               [function, total, hook(RANGEMAP.inclusion)]
+```
+
+### Range map choice
+
+You can get an arbitrarily chosen key of a `RangeMap` in O(1) time. The same
+key will always be returned for the same range map, but no guarantee is given
+that two different range maps will return the same element, even if they are
+similar.
+
+```k
+  syntax KItem ::= choice(RangeMap)                      [function, hook(RANGEMAP.choice), klabel(RangeMap:choice)]
+```
+
+### Implementation of range maps
+
+TBD
```

Comment:
If we're going to merge this into `domains.md`, we should have a more descriptive comment here explaining why the implementation is TBD.